tools/preprocess.py

```diff
@@ -11,47 +11,71 @@
 
 import argparse
 from pathlib import Path
-import os
+import re
 import tempfile
 import subprocess
 import sys
+import typing
 
 
 def fail(message):
     print(message, file=sys.stderr)
     sys.exit(1)
 
 
-def process_file(version, filename, input, output):
+def process_file(
+    version: str,
+    filename: str,
+    input: typing.TextIO,
+    output: typing.TextIO,
+):
     output.write(f'#line 1 "{filename}"\n')
+    # whether the current line follows a #pragma increment_block_number,
+    # including continuation lines (lines after a \-ending line)
+    in_pragma_incblocknum = False
+    # the line where the #pragma increment_block_number is
+    pragma_incblocknum_first_line_num = None
+    # all the lines from the #pragma increment_block_number line to the last
+    # continuation line, as a list[str]
+    pragma_incblocknum_lines = None
     for i, line in enumerate(input, start=1):
-        if line.startswith("#pragma increment_block_number"):
-            # Grab pragma argument (if any) and remove quotes
-            arg = line.strip()[len("#pragma increment_block_number ") + 1 : -1]
-            amount = 0
-            for part in arg.split():
-                kv = part.split(":")
-                if len(kv) != 2:
-                    fail(
-                        f"{filename}:{i}: increment_block_number must be followed by a list of version:amount pairs"
+        if not in_pragma_incblocknum and line.startswith(
+            "#pragma increment_block_number "
```

Comment:
```suggestion
            "#pragma increment_block_number"
```
it might be useful to only write `#pragma increment_block_number` and still have it be automatically updated (if you need to add a new one to a file). Ditto in fix_bss.py. Not sure though, what do you think?

Reply:
I put a space to avoid matching e.g. `increment_block_number_somerandomgarbage` but I guess that's not really a concern

Reply:
Yeah I'd rather be able to write a bare `increment_block_number`